pkg/skaffold/debug/transform.go

```diff
@@ -120,18 +159,58 @@ func transformPodSpec(metadata *metav1.ObjectMeta, podSpec *v1.PodSpec, retrieve
 	portAlloc := func(desiredPort int32) int32 {
 		return allocatePort(podSpec, desiredPort)
 	}
-	// containers are required to have unique name within a pod
+	// map of containers -> debugging configuration maps; k8s ensures that a pod's containers are uniquely named
 	configurations := make(map[string]map[string]interface{})
+	// the container images that require debugging support files
+	var containersRequiringSupport []*v1.Container
+	// the set of image IDs required to provide debugging support files
+	requiredSupportImages := make(map[string]bool)
 	for i := range podSpec.Containers {
 		container := &podSpec.Containers[i]
-		// we only reconfigure build artifacts
-		if configuration, err := transformContainer(container, retrieveImageConfiguration, portAlloc); err == nil {
+		// the usual retriever returns an error for non-build artifacts
+		imageConfig, err := retrieveImageConfiguration(container.Image)
+		if err != nil {
+			continue
+		}
+		// requiredImage, if not empty, is the image ID providing the debugging support files
+		if configuration, requiredImage, err := transformContainer(container, imageConfig, portAlloc); err == nil {
```

Comment:
I am not entirely sure about Go coding style, but should this if statement be separated like the one above for readability?

Reply:
It scopes the variables to the if and else blocks.

Reply:
There's a little bit of an explanation [here](https://golang.org/doc/effective_go.html#if).